generate_diagrams.py

```
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+"""Generate architecture diagrams for the project using various tools."""
+import ast
+import subprocess
+import sys
+from pathlib import Path
+
+
+def check_tool(tool_name):
```

Ruff (0.12.2)

Missing return type annotation for public function `check_tool`

Add return type annotation: `Optional[bool]`

(ANN201)